mimic/model/maas_objects.py

```diff
@@ -484,80 +599,174 @@ def __init__(self, clock):
         can be found in `the Rackspace Cloud Monitoring Developer Guide, appendix B
             <http://docs.rackspace.com/cm/api/v1.0/cm-devguide/content/appendix-check-types.html>`_
         """
+        self.agents = []
+        self.alarm_states = []
+
         self.check_types = {
             'agent.cpu': CheckType(clock=clock, metrics=[
-                Metric(name='user_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='wait_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='sys_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='idle_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='irq_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='usage_average', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='min_cpu_usage', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='max_cpu_usage', type=METRIC_TYPE_NUMBER, unit='percent'),
-                Metric(name='stolen_percent_average', type=METRIC_TYPE_NUMBER, unit='percent')]),
+                _agent_metric(name='user_percent_average', type=METRIC_TYPE_NUMBER, unit='percent'),
```

Comment:
It seems to me that this `__init__` function is growing out of control.  There aren't tests for every single one of these checks and metrics, and we seem to keep adding to them.  Is there any automated way we could keep all these metrics in sync with the actual public API, some data list we could be importing and parsing, rather than procedurally generating said list on every `__init__`?

If this really has to be manually constructed, it might be good to start grouping things by common features to reduce the line-count bloat; for example, in this case, something like:

```python
MetricListBuilder(type=METRIC_TYPE_NUMBER, unit='percent').metrics(
    'user_percent_average', 'wait_percent_average', 'sys_percent_average', # ...
)
```

we could potentially make this a lot shorter and less repetitive that way.